api/options.c

```diff
@@ -270,7 +270,7 @@ nlopt_result
 NLOPT_STDCALL nlopt_set_lower_bounds(nlopt_opt opt, const double *lb)
 {
      nlopt_unset_errmsg(opt);
-     if (opt && (opt->n == 0 || lb)) {
+     if (opt && (opt->n > 0) && lb) {
```

Comment:
This isn't right, because it will return `NLOPT_INVALID_ARGS` if `n == 0`, when actually that is not an error.

The code here is actually correct (it only allows a NULL `lb` in cases where `n == 0`, in which case nothing is written), so all we need to do is to silence the warning.   Couldn't that be accomplished by doing `lb && memcpy(...)` in place of `memcpy(...)`?